### DISCUSSION TITLE
Result of `ClassPropertyAssignToConstructorPromotionRector` to multiple lines?

### DISCUSSION THREAD

**Discussion by dakur** [rectorphp/rector#7361] [Q&A] [answered]
Hi, I'm sorry if I missed something but I'm not able to refactor normal properties to promoted ones. It squashes all promoted props into one line. What do I do wrong? There is no configuration or reference to resolve this in the [rule docs](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#classpropertyassigntoconstructorpromotionrector). Thank you.

**rector.php**

```php
use Rector\Config\RectorConfig;
use Rector\Core\Configuration\Option;

return static function (RectorConfig $rectorConfig): void {
	$rectorConfig->paths([
		__DIR__ . '/src'
	]);

	$rectorConfig->importNames(true); // use "use" statements instead of FQN
	$rectorConfig->importShortClasses(false); // don't import root classes (e.g. \LogicException)
	$rectorConfig->parameters()->set(Option::APPLY_AUTO_IMPORT_NAMES_ON_CHANGED_FILES_ONLY, true); // don't import everything, only affected

	$rectorConfig->rule(\Rector\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector::class);
};
```

**Code to refactor**

```php
final class Whatever
{
    private int $a;
    private string $b;

    public function __construct(
         int $a,
         string $b
    ) {
        $this->a = $a;
        $this->b = $b;
    }
}
```

**Expected**

```php
final class Whatever
{
    public function __construct(
         private int $a,
         private string $b
    ) {}
}
```

**Actual**

```php
final class Whatever
{
    public function __construct(private int $a, private string $b) {}
}
```

Screenshot from PhpStorm:

![image](https://user-images.githubusercontent.com/9862581/184286447-f93b537f-fb4a-4724-811b-36f595ef5115.png)

**Comment by samsonasik** [comment, marked as answer]
That is expected, ref https://github.com/rectorphp/rector#how-to-apply-coding-standards , you may need coding standard tool for that

**Reply by dakur** [reply]
Thank you very much for the explanation, I couldn't find it there at the bottom of the readme. 

While I understand the technical issue, from user perspective, the behavior is very confusing because when refactoring, I expect that the tool preserve what I already have there. So I think it would be very useful that newcomer is warned about that behavior when reading about setting up rector (Configuration section probably). What do you think?
